common/djangoapps/third_party_auth/samlproviderconfig/views.py

```diff
@@ -0,0 +1,56 @@
+"""
+    Viewset for auth/saml/v0/samlproviderconfig
+"""
+
+from edx_rbac.mixins import PermissionRequiredMixin
+from edx_rest_framework_extensions.auth.jwt.authentication import JwtAuthentication
+from enterprise.models import EnterpriseCustomerIdentityProvider
+from rest_framework import permissions, viewsets
+from rest_framework.authentication import SessionAuthentication
+
+from third_party_auth.samlutils.utils import fetch_enterprise_customer_by_id
+
+from ..models import SAMLProviderConfig
+from .serializers import SAMLProviderConfigSerializer
+
+
+class SAMLProviderMixin(object):
+    authentication_classes = [JwtAuthentication, SessionAuthentication]
+    permission_classes = [permissions.IsAuthenticated]
+    serializer_class = SAMLProviderConfigSerializer
+
+
+class SAMLProviderConfigViewSet(PermissionRequiredMixin, SAMLProviderMixin, viewsets.ModelViewSet):
+    """
+    A View to handle SAMLProviderConfig CRUD
+
+    Usage:
+        [HttpVerb] /auth/saml/v0/providerconfig/?enterprise-id=uuid
+
+    permission_required refers to the rule name defined in edx-rbac
```

Comment:
"`permission_required` refers to a Django permission defined in `enterprise.rules`." is a more accurate way to phrase this first sentence.